src/main/java/tr/com/srdc/cda2fhir/jolt/ResourceAccumulator.java

```diff
@@ -36,25 +36,90 @@ public Object transform(Object input, Map<String, Object> context) {
 			return null;
 		}
 		Map<String, Object> resource = (Map<String, Object>) input;
+
 		if (resource == null) {
 			resource = new LinkedHashMap<String, Object>();
 		}
+
+		List<Object> resources = (List<Object>) context.get("Resources");
+		if (resources == null) {
+			resources = new ArrayList<Object>();
+			context.put("Resources", resources);
+		}
+
+		int id = resources.size() + 1;
+
+		String reference = String.format("%s/%s", resourceType, id);
+		String display = AdditionalModifier.getDisplay(resource);
+
+		Map<String, Object> resourceMap = (Map<String, Object>) context.get("RESOURCE_MAP");
+
+		if (resourceMap == null) {
+			resourceMap = new HashMap<String, Object>();
+			context.put("RESOURCE_MAP", resourceMap);
+		}
+
+		List<Object> identifiers;
+		if (resource.get("identifier") instanceof LinkedHashMap) {
+			identifiers = new ArrayList<Object>();
+			identifiers.add(resource.get("identifier"));
+		} else {
+			identifiers = (List<Object>) resource.get("identifier");
+
+		}
+
 		if (input != null && "Medication".equals(resourceType)) {
-			CodeableConceptMap medicationMap = (CodeableConceptMap) context.get("MEDICATION_MAP");
-			if (medicationMap != null) {
-				Map<String, Object> existing = medicationMap.get(resource.get("code"));
-				if (existing != null) {
-					return existing;
+			MedicationMap medicationMap = (MedicationMap) context.get("MEDICATION_MAP");
+			Map<String, Map<String, Object>> referenceOrgMap = (Map<String, Map<String, Object>>) context
+					.get("REF_ORG_MAP");
+
+			if (medicationMap == null) {
+				medicationMap = new MedicationMap();
+				context.put("MEDICATION_MAP", medicationMap);
+			}
+
+			if (referenceOrgMap == null) {
+				referenceOrgMap = new HashMap<String, Map<String, Object>>();
+				context.put("REF_ORG_MAP", referenceOrgMap);
+			}
+
+			Map<String, String> manufacturer = (Map<String, String>) resource.get("manufacturer");
+			List<Object> orgIdentifiers = null;
+
+			if (manufacturer != null) {
+				String manuRef = manufacturer.get("reference");
+				Map<String, Object> organization = referenceOrgMap.get(manuRef);
+				if (organization != null) {
+					orgIdentifiers = (List<Object>) resource.get("identifier");
 				}
 			}
+			Map<String, Object> existing = medicationMap.get(resource, orgIdentifiers);
+			if (existing != null) {
+				return existing;
+			} else {
+				medicationMap.put(resource, orgIdentifiers);
+			}
 		}
 
-		if (input != null && "Organization".equals(resourceType)) {
+		if (input != null && "Organization".equals(resourceType))
+
+		{
+
 			IdentifierMap<Map<String, Object>> organizationMap = (IdentifierMap<Map<String, Object>>) context
 					.get("ORGANIZATION_MAP");
+			Map<String, Map<String, Object>> referenceOrgMap = (Map<String, Map<String, Object>>) context
```

Comment:
I do not think you need this. There is already a RESOURCE_MAP.

Reply:
yeah I'll change it.